code/game/gamemodes/revolution/revolution.dm

```diff
@@ -190,12 +181,28 @@
 				if(ROLE_REV in khrushchev.current.client.prefs.be_special)
 					promotable_revs += khrushchev
 		if(promotable_revs.len)
+			var/datum/mind/valid_head = locate() in head_revolutionaries
 			var/datum/mind/stalin = pick(promotable_revs)
-			revolutionaries -= stalin
-			head_revolutionaries += stalin
+			if(head_revolutionaries.len>0)
```

Comment:
I would go with: `if(length(head_revolutionaries))`

Reply:
Necro'ing here - so this is one of the differences between .len and length()? 

Although .... if head_revolutionaries.len were equal to 0, wouldn't that trigger as 0 which would read as FALSE?

Reply:
1) `length()` is just faster then `.len`, otherwise they're basically the same.

2) `if(length(head_revolutionaries))` is the equivalent to saying `if(length of head_revolutionaries is a non 0)` (yes this will pass if the number is negative). You can also go with:
`if(length(head_revolutionaries) > 0)` for clarity if you want but its optional.

Reply:
Oh crepes. I should be using `length()` then. I used `.len` a lot in my code recently.

I like the `> 0` as I'm all for clarity. I was curious about the neat "loopholes" and "workarounds" in the code.